mapper-pojo/src/main/java/org/hibernate/search/v6poc/entity/pojo/dirtiness/building/impl/PojoImplicitReindexingResolverBuildingHelper.java

```diff
@@ -122,4 +129,32 @@ boolean isEntity(PojoRawTypeModel<?> typeModel) {
 			BoundContainerValueExtractorPath<T, V> boundExtractorPath) {
 		return extractorBinder.create( boundExtractorPath );
 	}
+
+	ReindexOnUpdate getReindexOnUpdate(ReindexOnUpdate parentReindexOnUpdate,
+			PojoTypeModel<?> typeModel, String propertyName, ContainerValueExtractorPath extractorPath) {
+		if ( ReindexOnUpdate.NO.equals( parentReindexOnUpdate ) ) {
+			return ReindexOnUpdate.NO;
+		}
```

Comment:
Wondering if we shouldn't warn the user if the parent doesn't trigger reindex but some children is marked for it.

At least I understood this condition as "if the parent doesn't trigger reindex, we ignore the settings of the children".

Reply:
You understood it right.

From what I understood, warnings that cannot be removed and are triggered by a legitimate use case are a no-no in WildFly/JBoss EAP. Maybe `INFO`?

Also, knowing that the default is to include everything, and setting something to `NO` requires explicit action from the user, I wonder if such a warning would be useful. Consider the following model:

```
@Entity
@Indexed
public class A {
    @Id
    private Integer id;
    @OneToOne
    @IndexedEmbedded
    @IndexingDependency(reindexOnUpdate = ReindexOnUpdate.NO)
    private B b;
}
@Entity
public class B {
    @OneToOne(mappedBy = "b")
    private A a;
    @Field
    private String field;
}
```

The user explicitly said "I don't want changes to A.b to trigger reindexing of A".

When `A.b` changes, potentially the value `A.b.field` may change, but we already established that such a change should not trigger reindexing. Why should a change to `A.b.field` trigger reindexing when `A.b` didn't change, but not when `A.b` changed?

I'm honestly asking; there may be a reasoning that would make people expect that, I just don't see it.

Reply:
I think I wasn't clear. My issue is if the element is marked with `ReindexOnUpdate.NO` at some point but somewhere in the children, the user explicitly set it to `ReindexOnUpdate.YES`, which, if I understood correctly what you did, wouldn't change a thing.

Reply:
So, something like this?


```
@Entity
@Indexed
public class A {
    @Id
    private Integer id;
    @OneToOne
    @IndexedEmbedded
    @IndexingDependency(reindexOnUpdate = ReindexOnUpdate.NO)
    private B b;
}
@Entity
public class B {
    @OneToOne(mappedBy = "b")
    private A a;
    @Field
    @IndexingDependency(reindexOnUpdate = ReindexOnUpdate.USED)
    private String field;
}
```

It would work the same as my previous example, indeed, since `USED` (`YES`, if you will) is the default.

Would you log something when there's an explicit `USED` (`YES`), but not when there's no explicit configuration?

Reply:
That was my question.

On one hand, the user might be surprised that his explicit `USED` is ignored, on the other hand, maybe he will have it set to `YES` for another hierarchy so he might be annoyed by the warning.

Reply:
So... In this case, the users shouldn't even be using `reindexOnUpdate = ReindexOnUpdate.USED`, because it's the default, so this annotation is completely useless.

I could change the model, and offer a `@NoReindexOnUpdate` annotation instead, or `@IndexingDependency(ignore = true)`. That would prevent any kind of misuse. However:

1. I intended to use the same `@IndexedDependency` for defining the properties a transient property is based on; something like `@IndexedDependency(derivedFrom = {@ObjectPath(...), @ObjectPath(...)})`
2. I expected to provide, at some point in the future, an alternative `ReindexOnUpdate.SHALLOW` value, which would trigger reindexing for any value within the same entity, but never for a value in another entity that would require us to walk through associations to get to the entity to reindex.
3. I suspect that, at some point, we'll get people that want updates to all properties to be ignored, unless they explicitly tell us the opposite. In which case a value of YES could come handy.

I guess 1. could be solved by just relying on a different property for that (`@Derived(from = (derivedFrom = {@ObjectPath(...), @ObjectPath(...)})`.
2. and 3. would require to either offer one annotation per mode (`@NoReindexOnUpdate`, `@ReindexOnShallowUpdate`, `@ReindexOnUpdate`), or to use the current solution of exposing an annotation with an enum attribute.

Maybe I could simply rename `YES`/`USED` to `DEFAULT`? That would make it clear that it can't be forced, it's just the default. And later we could add other non-default modes such as `YES`/`USED` (reindex if the property is used, even if the default for the entity is `NO`) or `SHALLOW`.

Reply:
Renamed `USED` to `DEFAULT` and force-pushed as per our discussion on HipChat. Will merge as soon as CI goes green.